### PR TITLE
Fix sqlinstance user labels to filter out KRM-style labels

### PR DESCRIPTION
**Pull request by jasonvigil**
Fixes issue reported in https://github.com/GoogleCloudPlatform/k8s-config-connector/issues/2706#issuecomment-2357595555

### CODE REVIEW

**Comment by jasonvigil** [comment]
CC @maqiuyujoyce --

Note, the [Cleanup duplicate label helper functions](https://github.com/GoogleCloudPlatform/k8s-config-connector/pull/2808/commits/6591d25244fc7b86f202e788607253e9254ff028) refactor will cause a slight behavior change when exporting resources into terraform HCL format. Now, the exported HCL will include the "managed-by-cnrm" label. I think this is ok, as long as the export is only used for debugging purposes, and users do not terraform apply the exported HCL (and therefore potentially cause confusion when resource labels incorrectly report they are managed by cnrm when, in
fact, they are not managed by cnrm). It could be argued this new behavior is "more correct", because the exported HCL now exactly matches what KCC is applying.

But, want to get your opinion on this refactor. Can you PTAL?

File: pkg/label/label_test.go [hunk 1]
```diff
@@ -48,56 +31,44 @@ func TestNilMap(t *testing.T) {
...
 		"managed-by-cnrm": "true",
 	}
 	if !reflect.DeepEqual(result, expectedResult) {
 		t.Errorf("results mismatch: got '%v', want '%v'", result, expectedResult)
 	}
-	result = label.NewGCPLabelsFromK8SLabels(labels)
-	expectedResult = map[string]interface{}{
-		"key1": "val1",
-		"key2": "val2",
+	result = label.NewGCPLabelsFromK8sLabels(labels)
```
Comment (maqiuyujoyce) [inline pkg/label/label_test.go:49]: Nit: Looks like  all the rest are duplicated test cases. Could you remove them?

**Review by maqiuyujoyce — COMMENTED** [review]
Thank you for the fix and cleaning up the code, @jasonvigil ! Just one comment about unit test but feel free to unhold and address it in a separate PR if you prefer.

/lgtm
/approve
/hold

File: pkg/label/label_test.go (hunk 1 above)
Comment (jasonvigil) [reply pkg/label/label_test.go]: Ah, yeah, good catch! Thanks. Fixed

**Review by maqiuyujoyce — APPROVED** [review]
/lgtm
/approve

**Pull request merged**